student.rb

```diff
@@ -1,15 +1,10 @@
-class Student
-  attr_reader :age, :phase
-  attr_accessor :name
+require_relative 'teacher_student_class'
+
+class Student < TeacherStudent
```

Comment:
So, a `Student` is a `TeacherStudent`?  I'd go with `Person` or maybe `DBCPerson` here.